xapian-core/include/xapian/weight.h

```diff
@@ -468,7 +468,7 @@ class XAPIAN_VISIBILITY_DEFAULT TfIdfWeight : public Weight {
        the desired result will not be achieved. */
   public:
     /** Wdf normalizations. */
-    enum class WDF_NORM : int {
+    enum class wdfn_type : int {
```

Comment:
Not sure I like `wdfn_type`, etc - we don't have any other type names that end `_type` in the API currently (except for sub-types like `Xapian::TermIterator::value_type` which are required for compatibility with the C++ STL).

Reply:
Also, `wdfn_type::BOOLEAN`, etc seem less clear than `wdf_norm::BOOLEAN`.

Reply:
I have now changed the class name to wdf_norm, etc. The parameter names are now wdf_norm_, etc.